### PR TITLE
FIX: Verification of variables

### LINKED ISSUE #28622: Undefined array key 0
### Bug

Undefined array key 0 in htdocs/core/class/CMailFile.class.php on line 317

### Dolibarr Version

19.0

### Environment PHP

8.1

### Environment Database

MySQL

### Steps to reproduce the behavior and expected behavior

When the user wants to send a document by an e-mail (bill, proposal...).

### Attached files

_No response_

### PR DESCRIPTION
**Pull request by JeromeBARBOTIN**
Verification of the variables setting before its use.
Fix for #28622

### CODE REVIEW

File: htdocs/comm/propal/class/propal.class.php [hunk 1]
```diff
@@ -2816,7 +2816,7 @@ public function classifyBilled(User $user, $notrigger = 0, $note = '')
 		}
 
 		if (!$error) {
-			$modelpdf = $conf->global->PROPALE_ADDON_PDF_ODT_CLOSED ? $conf->global->PROPALE_ADDON_PDF_ODT_CLOSED : $this->model_pdf;
+			$modelpdf = isset($conf->global->PROPALE_ADDON_PDF_ODT_CLOSED) ? $conf->global->PROPALE_ADDON_PDF_ODT_CLOSED : $this->model_pdf;
```
Comment (frederic34) [inline htdocs/comm/propal/class/propal.class.php:2819]: you can write `$modelpdf = getDolUserString('PROPALE_ADDON_PDF_ODT_CLOSED', $this->model_pdf);`

File: htdocs/core/actions_sendmails.inc.php [hunk 2]
```diff
@@ -434,10 +434,10 @@
 					setEventMessages($mesg, null, 'mesgs');
 
 					$moreparam = '';
-					if (isset($paramname2) || isset($paramval2)) {
+					if (isset($paramname2) && isset($paramval2)) {
```
Comment (eldy) [inline htdocs/core/actions_sendmails.inc.php:437]: Why changing the || into &&  ?

If $paramval2 is set and not $paramname2, we should get '&mid='.$paramval2

Comment (JeromeBARBOTIN) [reply htdocs/core/actions_sendmails.inc.php:437]: wit the `||`, you allow to call `$paramname2` even if it is not defined. This lead to a php error (version 19.0.0):
screenshot

We could also remove  `|| isset($paramval2)`

Comment (eldy) [reply htdocs/core/actions_sendmails.inc.php:437]: So this means we must replace
`($paramname2 ? $paramname2 : 'mid')`
with
`(!empty($paramname2) ? $paramname2 : 'mid')`

Comment (JeromeBARBOTIN) [reply htdocs/core/actions_sendmails.inc.php:437]: ok, if `empty($paramname2)`does not return an error when `$paramname2`is not set.
This is one of the main reasons I have a lot of crash with Dolibarr 19.0.0 with php 8.1. 
[https://wiki.php.net/rfc/undefined_property_error_promotion](php documentation)

Comment (JeromeBARBOTIN) [reply htdocs/core/actions_sendmails.inc.php]: I replaced with `$paramname ?? 'id'`. I tested and it worked.

**Pull request merged**